native/screens/SignUp.jsx

```diff
@@ -1,180 +1,309 @@
-// import { useState } from "react";
-// import { useNavigate, Link } from "react-router-dom";
+import { useState, createRef } from 'react';
+import { useDispatch } from "react-redux";
+import { signUpUser } from "../services/redux/actions/uiActions.js";
+import { checkEmailAuth, verify } from '../services/api/users.js';
 
-// // assets
-// import { useDispatch } from "react-redux";
-// import { signUpUser } from "../../services/redux/slices/uiActions.js";
-// import { GenericModal } from "../components/Modal/GenericModal.jsx";
-// import "./SignUp.scss";
-// import { checkEmailAuth, verify } from "../../services/api/users.js";
-// import { handleChange } from "../../services/utils/formHandlers";
-// import { SingleActionButton } from "../components/Button/SingleActionButton.jsx";
-// import { DoubleActionModal } from "../components/Modal/DoubleActionModal.jsx";
-import { StyleSheet, Text, View } from "react-native";
+// Native Components
+import { 
+  StyleSheet, 
+  Text, 
+  TextInput, 
+  View, 
+  TouchableOpacity, 
+  Alert, 
+  Modal 
+} from "react-native";
 
-export const SignUp = () => {
-  return (
-    <View>
-      <Text>Sign up</Text>
-    </View>
-  );
-};
 
-// export const ReactSignUp = () => {
-//   const navigate = useNavigate();
-//   const dispatch = useDispatch();
-//   const [showModal, setShowModal] = useState(false);
-//   const [showSuccessModal, setShowSuccessModal] = useState(false);
-//   const [modalError, setModalError] = useState("");
-//   const [emailError, setEmailError] = useState(null);
-//   const [newUser, setNewUser] = useState({
-//     confirm_password: "",
-//     email: "",
-//     first_name: "",
-//     last_name: "",
-//     password: "",
-//   });
+export const SignUp = ({ navigation }) => {
+  // redux
+  const dispatch = useDispatch();
+  // state
+  const [emailError, setEmailError] = useState(null);
+  const [successModalVisible, setSuccessModalVisible] = useState(false);
+  const [newUser, setNewUser] = useState({
+    first_name: '',
+    last_name: '',
+    email: '',
+    password: '',
+    confirm_password: '',
+  });
+  // input references
+  const { 
+    firstNameInputRef, 
+    lastNameInputRef, 
+    emailInputRef, 
+    passwordInputRef, 
+    confirmPasswordInputRef } = createRef();
```

Comment:
i have not touched on refs at all yet, i'd love to schedule a 1:1 with you whenever you have time to teach me :)